### PR TITLE
[ALTO] Suppress and fix sonar related issues

### PR DESCRIPTION
**Pull request by ramizdundar**
I have suppressed most and fixed some sonar related issues
according to feedback (https://github.com/hazelcast/hazelcast/pull/23720
and Slack DMs) I get from @pveentjer.

Also see: https://hazelcast.slack.com/archives/C0356BG4R/p1677074927605439

### CODE REVIEW

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AbstractAsyncSocket.java [hunk 1]
```diff
@@ -137,21 +137,17 @@ public final void close(String reason, Throwable cause) {
         this.closeReason = reason;
         this.closeCause = cause;
 
-        if (cause == null) {
```
Comment (pveentjer) [inline hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AbstractAsyncSocket.java:140]: The original code structure I find a lot easier to read. If you agree with this, then such a refactoring makes the code worse, not better.

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAsyncSocket.java [hunk 2]
```diff
@@ -268,16 +269,6 @@ public void flush() {
         }
     }
 
-    private void resetFlushed() {
```
Comment (pveentjer) [inline hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAsyncSocket.java:271]: Can you restore the original code. This function is part of the NioAsyncSocket; not of the handler.

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/util/OS.java [hunk 3]
```diff
@@ -30,7 +30,7 @@ public final class OS {
     private static final int PAGE_SIZE = UnsafeLocator.UNSAFE.pageSize();
     private static final String OS_ARCH = System.getProperty("os.arch", "?");
     private static final boolean IS_64BIT = is64bit0(OS_ARCH);
-    private static final boolean IS_X86_64 = OS_ARCH.equals("amd64");
```
Comment (pveentjer) [inline hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/util/OS.java:33]: I'm happy with X86_64. AMD64 is more of a legacy name.

File: hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/AsyncSocketBuilderTest.java [hunk 4]
```diff
@@ -63,7 +63,8 @@ public void test_setReadHandler_whenAlreadyBuild() {
         builder.setReadHandler(new DevNullReadHandler());
         builder.build();
 
-        assertThrows(IllegalStateException.class, () -> builder.setReadHandler(new DevNullReadHandler()));
```
Comment (pveentjer) [inline hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/AsyncSocketBuilderTest.java:66]: What is the advantage here?

File: hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/util/JVMTest.java [hunk 5]
```diff
@@ -25,13 +25,13 @@ public class JVMTest {
...
         System.out.println(JVM.getMajorVersion());
-        assertEquals(JVM.parseVersionString("1.6.0_23"), 6);
-        assertEquals(JVM.parseVersionString("1.7.0"), 7);
-        assertEquals(JVM.parseVersionString("1.7.0_80"), 7);
-        assertEquals(JVM.parseVersionString("1.8.0_211"), 8);
-        assertEquals(JVM.parseVersionString("9.0.1"), 9);
-        assertEquals(JVM.parseVersionString("11.0.4"), 11);
-        assertEquals(JVM.parseVersionString("12"), 12);
-        assertEquals(JVM.parseVersionString("12.0.1"), 12);
+        assertEquals(6, JVM.parseVersionString("1.6.0_23"));
```
Comment (pveentjer) [inline hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/util/JVMTest.java:28]: I always keep forgetting the order.

File: hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java [hunk 6]
```diff
@@ -284,7 +279,6 @@ public void shutdown() {
 
         logger.info("TcpBootstrap shutdown");
 
-        shutdown = true;
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java:287]: What happened this this field?

File: hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/AsyncSocketBuilderTest.java (hunk 4 above)
Comment (ramizdundar) [reply hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/AsyncSocketBuilderTest.java:66]: Sonar says there should be one function call inside the lambda that can throw for better testing. Hence I extracted constructor.

File: hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java (hunk 6 above)
Comment (ramizdundar) [reply hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java:287]: It was unused. We can add shutdown field when we are going to implement functionality. WDYT?

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AbstractAsyncSocket.java (hunk 1 above)
Comment (pveentjer) [reply hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AbstractAsyncSocket.java:140]: I prefer the original one.

```
 if (cause == null) {
            if (logger.isInfoEnabled()) {
                if (reason == null) {
                    logger.info("Closing  " + this);
                } else {
                    logger.info("Closing " + this + " due to " + reason);
                }
            }
        } else {
            if (logger.isWarningEnabled()) {
                if (reason == null) {
                    logger.warning("Closing  " + this, cause);
                } else {
                    logger.warning("Closing " + this + " due to " + reason, cause);
                }
            }
        }
        ```
        Vs
        
        ```
        if (cause == null && logger.isInfoEnabled()) {
            if (reason == null) {
                logger.info("Closing  " + this);
            } else {
                logger.info("Closing " + this + " due to " + reason);
            }
        } else if (logger.isWarningEnabled()) {
            if (reason == null) {
                logger.warning("Closing  " + this, cause);
            } else {
                logger.warning("Closing " + this + " due to " + reason, cause);
            }
        }
        ```
        
        Perhaps I'm an old fart?

File: hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/AsyncSocketBuilderTest.java (hunk 4 above)
Comment (pveentjer) [reply hazelcast-tpc-engine/src/test/java/com/hazelcast/internal/tpc/AsyncSocketBuilderTest.java:66]: Ok.

File: hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java (hunk 6 above)
Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java:287]: Remove what is unused.

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AbstractAsyncSocket.java (hunk 1 above)
Comment (ramizdundar) [reply hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AbstractAsyncSocket.java]: Reverted https://github.com/hazelcast/hazelcast/pull/23740/commits/7483a2aca1f78971e2c5efd0d99d1d8b2dcbd201

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAsyncSocket.java (hunk 2 above)
Comment (ramizdundar) [reply hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAsyncSocket.java:271]: Reverted https://github.com/hazelcast/hazelcast/pull/23740/commits/7483a2aca1f78971e2c5efd0d99d1d8b2dcbd201

**Review by mdumandag — APPROVED** [review]

File: hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAsyncSocket.java [hunk 7]
```diff
@@ -268,13 +269,13 @@ public void flush() {
         }
     }
 
+    @SuppressWarnings("java:S3398")
     private void resetFlushed() {
         flushThread.set(null);
 
-        if (!unflushedBufs.isEmpty()) {
```
Comment (pveentjer) [inline hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAsyncSocket.java:274]: Can you restore this. I'm a simple person and like simple code.

Comment (ramizdundar) [reply hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAsyncSocket.java]: https://github.com/hazelcast/hazelcast/pull/23740/commits/2daff84fb3005813adebe904f51db834144725f7

**Review by pveentjer — APPROVED** [review]

**Pull request merged**